tools/lsp/server_loop.rs

```diff
@@ -175,22 +87,10 @@ impl DocumentCache {
     }
 }
 
-#[cfg(feature = "preview-api")]
-pub struct PreviewApi {
-    pub highlight: Box<
-        dyn Fn(
-            &Rc<Context>,
-            Option<std::path::PathBuf>,
-            u32,
-        ) -> Result<(), Box<dyn std::error::Error>>,
-    >,
-}
-
 pub struct Context {
-    pub document_cache: RefCell<DocumentCache>,
+    pub document_cache: RefCell<crate::server_loop::DocumentCache>,
     pub server_notifier: crate::ServerNotifier,
     pub init_param: InitializeParams,
-    #[cfg(feature = "preview-api")]
     pub preview: PreviewApi,
```

Comment:
```suggestion
    pub preview: Box<dyn PreviewApi>,
```

Alternative,  move that `preview` in the ServerNotifier.    Or simply implement PreviewApi trait for `ServerNotifier`